### PR TITLE
Rate refactor

### PR DESCRIPTION
**Pull request by samlown**
* Major refactor of Tax Rate handling
* Tags outside of the invoice are now gone and moved to Rates.
* `org.Item` now has it's own key field (required for TicketBAI)
* `cbc.Code` now supports adding `.`
* Moved from `cbc.Meta` to `cbc.CodeSet` for keys used potentially by regime specific conversion tools.
* Removed `Base` from `tax.CategoryTotal`

### CODE REVIEW

File: cbc/key.go [hunk 1]
```diff
@@ -54,14 +85,42 @@ func (k Key) In(set ...Key) bool {
 	return false
 }
 
+// IsEmpty returns true if the key has no value.
+func (k Key) IsEmpty() bool {
+	return k == KeyEmpty
+}
+
+// HasValidKeyIn provides a validator to check the Key's
+// value is within the provided know set.
```
Comment (lucanioi) [inline cbc/key.go:94]: know set?

File: regimes/it/invoice_validator.go
Comment (lucanioi) [inline regimes/it/invoice_validator.go:1]: what's going on here?

File: regimes/it/pay.go
Comment (lucanioi) [inline regimes/it/pay.go:1]: Nice, I think this approach is much more consistent with how we handle other codes!!

File: regimes/it/scenarios.go [hunk 2]
```diff
@@ -17,89 +80,141 @@ var invoiceScenarios = &tax.ScenarioSet{
 		// **** TIPO DOCUMENTO ****
 		{
 			Types: []cbc.Key{bill.InvoiceTypeStandard},
-			Meta: cbc.Meta{
+			Name: i18n.String{
+				i18n.EN: "Regular Invoice",
+				i18n.IT: "Fattura",
+			},
+			Codes: cbc.CodeSet{
```
Comment (lucanioi) [inline regimes/it/scenarios.go:87]: This makes sense for Italy, I just hope it translates well to other regimes in the future.

**Review by lucanioi — APPROVED** [review]
Looks great, I'm looking forward to using this in the XML converter. Only thing is it seems like this doesn't include latest changes to the code (like `it/invoices.go` being replaced by `it.invoice_validator.go`

File: regimes/it/invoice_validator.go
Comment (samlown) [reply regimes/it/invoice_validator.go:1]: I have no idea... adding this placeholder file was the only way I could get the linter to work!

File: regimes/pt/tags.go [hunk 3]
```diff
@@ -7,46 +7,12 @@ import (
...
-	TagNonTaxable    cbc.Key = "non-taxable"
-)
-
 // Invoice type tags
 const (
 	TagInvoiceReceipt cbc.Key = "invoice-receipt"
 )
 
-var invoiceTags = []*tax.Tag{
+var invoiceTags = []*tax.KeyDefinition{
```
Comment (cavalle) [inline regimes/pt/tags.go:15]: @samlown I noticed that, in IT, you moved the `invoiceTags` to `scenarios.go` and deleted `tags.go`. Do we want to do the same in PT?

Comment (samlown) [reply regimes/pt/tags.go:15]: yes, that does make sense... moving.

**Pull request merged**